### ISSUE TITLE
Passing content length when uploading stream?

### ISSUE THREAD

**Issue by wmertens** [pnp/pnpjs#2218]
### Category
- Question

### Version

Please specify what version of the library you are using: [     v3    ]

Please specify what version(s) of SharePoint you are targeting: [   online    ]

### Expected / Desired Behavior / Question

in v1/v2, we used to call ``.configure({headers: {'Content-Length': `${size}`}})`` before calling `.addChunked`, and it would add a content-length header, as per the example [here](https://github.com/pnp/pnpjs/blob/f724231148c4b221acb190c11d1795934dab7719/docs/v2/sp/files/index.html#L2593). Now, `.configure` is not available and I wonder how to add such a header.

Is the solution to create a separate single-use spfi() stack that includes the length header?

Or is there another way to configure the next fetch?

The [current example](https://github.com/pnp/pnpjs/blob/c0b3cb0e6b8f7b9f0272f238ea04f9784de5457f/docs/sp/files.md?plain=1#L128) retains the v2 comment but doesn't add the header.

**Comment by wmertens** [comment]
Ah I just found https://github.com/pnp/pnpjs/discussions/1954#discussioncomment-1835938 - I'll test and and update the example if it works

**Comment by wmertens** [comment]
I'm having trouble with the upload - the fetch doesn't get a body in the `init` it gets from the `send` moment, which I'm assuming means that the spPost is already posting without the body being added.
Bedtime now but pointers would be appreciated.

**Comment by wmertens** [comment]
Ah - ok so what happens is that when you use addChunked, it first POSTs `.../files/add(...)` which returns a GUID to upload to, and only then it starts uploading the file via `.../startUpload(...)` which gets the body.

So when using InjectHeaders, it puts the headers on the wrong request.

Furthermore, each fragment of the stream is uploaded separately after loading it in memory, so the Content-Length header would be wrong anyway and should be provided by File.startUpload and File.continueUpload themselves.

In other words, adding the length is not possible and needs an extra parameter to addChunked in node-js/behaviors so it can check if the stream length was correct.

Thoughts @juliemturner @patrick-rodgers?

**Comment by juliemturner** [comment]
Should work, but we'll look into it.

**Comment by juliemturner** [comment]
Ok, so I tested this by doing the following with a small file and a very large file and both worked without having to add any additional headers.  Note that the imports are somewhat different than the existing sample. 

```TypeScript
import { spfi } from "@pnp/sp";
import "@pnp/nodejs";
import "@pnp/sp/webs";
import "@pnp/sp/lists";
import "@pnp/sp/folders/list";
import "@pnp/sp/files/folder";
import { createReadStream } from 'fs';

// get a stream of an existing file
const stream = createReadStream("c:/temp/file.txt");

// now add the stream as a new file
const sp = spfi(...);

const fr = await sp.web.lists.getByTitle("Documents").rootFolder.files.addChunked( "new.txt", stream, undefined, true );
```
